### PR TITLE
hide temp tensor allocation

### PR DESCRIPTION
**Pull request by panyx0718**
test=develop

### CODE REVIEW

File: paddle/fluid/framework/operator.h [hunk 1]
```diff
@@ -368,6 +368,30 @@ class ExecutionContext {
     return op_.Outputs(name);
   }
 
+  template <typename T, typename DevContext>
+  Tensor AllocateTmpTensor(const framework::DDim& dim,
```
Comment (chengduoZH) [inline paddle/fluid/framework/operator.h:372]: `DevContext` can be removed.

File: paddle/fluid/operators/conv_op.h [hunk 2]
```diff
@@ -158,10 +157,7 @@ class GemmConvKernel : public framework::OpKernel<T> {
     // to call the matrix multiplication interface.
     Tensor col_matrix;
     if (is_expand) {
-      auto tmp_allocation_ptr =
-          platform::DeviceTemporaryAllocator::Instance().Get(dev_ctx).Allocate(
-              framework::product(col_shape) * sizeof(T));
-      col = framework::GetTensor<T>(std::move(tmp_allocation_ptr), col_shape);
+      col = context.AllocateTmpTensor<T, DeviceContext>(col_shape, dev_ctx);
```
Comment (chengduoZH) [inline paddle/fluid/operators/conv_op.h:160]: It is unnecessary that passing `DeviceContext` to `AllocateTmpTensor`.

File: paddle/fluid/framework/operator.h (hunk 1 above)
Comment (panyx0718) [reply paddle/fluid/framework/operator.h:372]: It's needed for `dev_ctx`.

File: paddle/fluid/operators/conv_op.h (hunk 2 above)
Comment (panyx0718) [reply paddle/fluid/operators/conv_op.h:160]: same as above

**Review by chengduoZH — APPROVED** [review]
LGTM

**Pull request merged**